### PR TITLE
compute: Replace Int64 with UInt64 for sys object

### PR DESCRIPTION
**Pull request by jkosh44**
### Motivation
This PR refactors existing code.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-protobuf` label.
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - N/A

### CODE REVIEW

File: test/sqllogictest/select_introspection.slt
Comment (jkosh44) [inline test/sqllogictest/select_introspection.slt:44]: This test is always breaking and I can never tell why, it's already extremely flaky and we do a bunch of hacky things to get it to pass. I just decided to delete it once and for all. If you feel strongly about not deleting it, then I will spend the time to try and fix it again.

**Comment by jkosh44** [comment]
This is a breaking change and needs to wait for the next release that allows breaking changes.

**Comment by jkosh44** [comment]
@antiguru I'm not sure who the right person is to review this, it's concerned mostly with built in introspection sources. Can you re-assign this if it's not you?

It's not urgent because it's a breaking change and needs to wait a little bit before merging (see above).

File: test/sqllogictest/select_introspection.slt
Comment (antiguru) [reply test/sqllogictest/select_introspection.slt:44]: Yes, this test is a nightmare to maintain. I'm OK with removing it.

**Review by antiguru — APPROVED** [review]
I think this looks good. I added some changes/optimizations in https://github.com/jkosh44/materialize/pull/1, feel free to consider it!

At the moment, I'm the right person to review, but I'll try to get more people ramped up.

**Comment by jkosh44** [comment]
> I think this looks good. I added some changes/optimizations in [jkosh44#1](https://github.com/jkosh44/materialize/pull/1), feel free to consider it!
> 
> At the moment, I'm the right person to review, but I'll try to get more people ramped up.

Thanks! I just merged your updated. I'll merge this once we are ready for a breaking change.

**Pull request merged**
